DESIGN.md

```diff
@@ -0,0 +1,370 @@
+# Apollo Android design discussion
+
+These are some preliminary ideas for mapping GraphQL query results to Java, based on what we currently do for Swift in Apollo iOS.
+
+## Query classes
+
+I would expect us to generate classes for queries (and other operations). Instances of these classes represent a particular query, fully configured with the required variables.
```

Comment:
Generating interfaces gives users more flexibility for plugging in their own implementation for the models (eg.: AutoValue, Immutables or even manually extend).
Also, following the composition over inheritance pattern, we could have a separate class that takes the interfaces and configures them for execution.